File: 6-kyu/31-Take-a-Ten-Minutes-Walk/go/main.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(IsValidWalk([]rune{'n', 's'}))
	fmt.Println(IsValidWalk([]rune{'n', 's', 'n', 's', 'n', 's', 'n', 's', 'n', 's'}))
	fmt.Println(IsValidWalk([]rune{'w', 'e', 'w', 'e', 'w', 'e', 'w', 'e', 'w', 'e', 'w', 'e'}))
}

// IsValidWalk :
func IsValidWalk(walk []rune) bool {
	if len(walk) != 10 {
		return false
	}
	counter := map[rune]int{}
	for i := 0; i < len(walk); i++ {
		counter[walk[i]] += 1
	}
	return (counter['n']-counter['s'] == 0) && (counter['w']-counter['e'] == 0)
}

///================ other practices ==================///

func IsValidWalkA(walk []rune) bool {
	if len(walk) != 10 {
		return false
	}
	var x, y int
	for _, r := range walk {
		switch r {
		case 's':
			y--
		case 'n':
			y++
		case 'w':
			x--
		case 'e':
			x++
		}
	}
	return x == 0 && y == 0
}

File: 6-kyu/31-Take-a-Ten-Minutes-Walk/python/main.py

def is_valid_walk(walk):
    if len(walk) != 10:
        return False
    counter = {'n':0,'s':0,'w':0,'e':0}
    i = 0
    while i<len(walk):
        counter[walk[i]] += 1
        i += 1
    if counter['n']-counter['s'] == 0 and counter['w']-counter['e'] == 0:
        return True
    return False

 
print(is_valid_walk(['n','s','n','s','n','s','n','s','n','s']))

### ================ other practices ==================

def isValidWalkA(walk):
    return len(walk) == 10 and walk.count('n') == walk.count('s') and walk.count('e') == walk.count('w')